parcels/kernel.py

```diff
@@ -115,14 +115,18 @@ def __init__(self, fieldset, ptype, pyfunc=None, funcname=None,
             self.vector_field_args = kernelgen.vector_field_args
             fieldset = self.fieldset
             for fname in self.vector_field_args:
-                f = getattr(fieldset, fname)
+                fields = fieldset.fields_TMP
+                found = False
+                for f in fields:
+                    if f.name == fname:
```

Comment:
I was thinking this was not efficient.
But on top of that, it's wrong, since fields can have same names. so this `vector_field_args` that works using name is not correct. This needs to be changed